proposals/1/patch-steps.md

```diff
@@ -0,0 +1,139 @@
+### Advantages
+
+- Format is in the "Steps" style that manual editing software may expect.
+
+- Format can be extended in future if issues are encountered.
+
+- Format follows a sequenced order that may be useful given the aforementioned expandability requirements. 
+
+### Disadvantages
+
+- Due to being in the "steps" style, the format is sequence-based, which makes removing sections require keeping track of the current position.
+
+- The format will still require hand-editing to import objects due to there being no reliable detection mechanism for these cases.
+
+- Making a modified object requires a blank object and a patch that first imports the source and then applies the modifications.
+
+- The format is more complicated than the previous one.
+
+## The Patch Format
+
+A Patch is an array of objects, known as PatchSteps.
+
+The "step" structure is used to allow existing editing tools that understand it to incorporate it with ease.
+
+(When defining a Step, try to ensure it can be defined in the form of a, well, Step, if possible.)
+
+And furthermore, as such, the "type" field describes the specific step type as with other types of Step.
+
+Further details are type-specific.
+
+When executing a Patch, some state is maintained.
+
+There is the Current Value, and a stack which backs up the Current Value to return to parent values.
+
+The final target of the Current Value doesn't matter - it's merely used as an editing cursor.
+
+### ENTER
+
+`ENTER` steps push the Current Value onto the stack, then replace the Current Value with a property of it, thus 'entering' that property.
+
+They have a single additional string property, "index".
+
+This property is a property name of the current object or array.
+
+JavaScript rules mean that "0" and 0 are equivalent, and thus this can access array elements.
+
+### EXIT
+
+`EXIT` steps pop a new Current Value off the stack, replacing it.
+
+This allows going back to a parent value to modify other sub-values.
+
+### SET_KEY
+
+`SET_KEY` steps have two additional properties:
+
+"index" (containing the property name)
+
+And optionally "content" (Arbitrary content value)
+
+The property in the Current Value is set to a copy of the content.
+
+If content is not provided, the property is deleted.
+
+Note that SET_KEY with a non-numeric index is valid on Arrays, but will probably end badly.
+
+### REMOVE_ARRAY_ELEMENT
+
+`REMOVE_ARRAY_ELEMENT` steps have a numeric property "index" (an Array.splice index).
+
+It removes the element at that index from the Current Value.
+
+### ADD_ARRAY_ELEMENT
+
+`ADD_ARRAY_ELEMENT` steps have an optional numeric property "index" (again, an Array.splice index)...
+
+... and an arbitrary value "content".
+
+A copy of the content is inserted into the Current Value at the index.
+
+If the index is not present, the content is appended.
+
+### IMPORT
+
+`IMPORT` steps have a string property "src", and optionally "index" (another string property) along with "path" (an array of keys).
```

Comment:
Maybe specify that src is CC root (or assets root?)

Reply:
Kind of a complex question by itself, since the format part's theoretically supposed to be partially 'independent' to what's loading "src"... I'll work out how to clarify this bit after I've dealt with standardized-mods

Reply:
...Wait, I actually did handle this in the spec - on the next line.
`JSON is loaded from the file specified with "src" (the value does *not* contain ig.root, that's prepended as usual).`